### PR TITLE
Cleanup and remove more files on uninstall

### PR DESCRIPTION
**Pull request by tiran**
* /etc/nsswitch.conf.ipabkp
* /etc/openldap/ldap.conf.ipabkp
* /var/lib/ipa/sysrestore/*
* /var/named/dyndb-ldap/ipa/

See: https://pagure.io/freeipa/issue/2694
Signed-off-by: Christian Heimes <cheimes@redhat.com>

### CODE REVIEW

File: ipaserver/install/server/install.py [hunk 1]
```diff
@@ -1150,6 +1150,11 @@ def uninstall(installer):
                      'but it may\n'
                      'mean your system hasn\'t be restored to its '
                      'pre-installation state.', SYSRESTORE_DIR_PATH)
+    else:
+        # cleanup remaining sysrestore artifacts but keep the directory
+        for basedir, dirnames, filenames in os.walk(SYSRESTORE_DIR_PATH):
+            for filename in filenames:
+                os.unlink(os.path.join(basedir, filename))
```
Comment (rcritten) [inline ipaserver/install/server/install.py]: I'm unsure about this bit. A few lines up a line is printed that files remain and then they are removed. Ideally all files are removed/restored during uninstallation. If there are any left over then that is the bug and cleaning them up just covers it up.

**Comment by tiran** [comment]
@rcritten I added a comment to the section.

The files are only removed when ``sysrestore.state`` contains no more states to restore. The loop is only executed, when ``has_state`` is false. In that case, ``sysrestore.state`` still exists, but only contains the hostname of the machine. 

In some cases, there might be other files left. For example, one test system had two copies of ``ntp.conf`` after first installation has failed.

**Comment by rcritten** [comment]
I agree on the point of left-over files but I think we should find the root cause of the issue and not cover it up by cleaning up the mess afterward. On the plus side if this is indeed related to mostly ntpd config files that problem will solve itself soon enough :-)

**Comment by tiran** [comment]
OK, you convinced me. The latest version just removes ``sysrestore.state`` after all state information has been removed from the file.

**Comment by tiran** [comment]
master:

* 64438f861947febe5ea10eda56ff89fd8900e80a Cleanup and remove more files on uninstall

**Pull request closed without merging**